core/templates/dev/head/pages/exploration_player/AnswerClassificationService.js

```diff
@@ -134,6 +136,26 @@ oppia.factory('AnswerClassificationService', [
 
         if (ruleBasedOutcomeIsDefault && interactionIsTrainable &&
             ENABLE_ML_CLASSIFIERS) {
+          if (ENABLE_TRAINING_DATA_CLASSIFICATION) {
+            var objectFactory = AnswerClassificationResultObjectFactory;
+            for (var i = 0; i < answerGroups.length; i++) {
+              for (var j = 0; j < answerGroups[i].rules.length; j++) {
+                var rule = answerGroups[i].rules[j];
+                if (rule.type === RULE_TYPE_CLASSIFIER) {
```

Comment:
I don't think this sort of classification will be part of answerGroups.rules? I think you want to check whether the interaction supports training data classification, then check whether this answer group has training data, then check whether the learner's answer matches anything in the training data. 

I also looked at the exploration schema for AnswerGroup. I think we need to add a new field so that it stores a list of trained data? Since that list is explicitly edited by creators?

Reply:
> I don't think this sort of classification will be part of answerGroups.rules? I think you want to check whether the interaction supports training data classification, then check whether this answer group has training data, then check whether the learner's answer matches anything in the training data.

The training data of an answer group is actually stored within that answer group in a rule whose type is `RULE_TYPE_CLASSIFIER`. Check the test I've written in spec file. Answer group stores training data as a part of a rule.